### ISSUE TITLE
Postpone with server+client views.

### ISSUE THREAD

**Issue by bogdanudrescu** [vaadin/flow#8135]
Trying https://github.com/vaadin/flow/issues/3619 on mixed client and server side view from @manolo https://github.com/bogdanudrescu/flow-mixed adding `BeforeLeaveEvent` `postpone` and `proceed`.

Test 1 (check again after #8170 is merged):
1. click on Server View2 tab
2. click on Server View1 tab
-> question open
  - choose YES
  -> page view updates (expected: URL updates as well)
  - choose NO
  -> selected tab updates, URL stays the same (expected: no update to the page)
  - choose NO, then click on Server View2 tab
  -> question is asked again (expected: nothing should happen)

Test 2:
1. click on Server View2 tab
2. click on Client View tab
-> question open
  - choose YES
  -> page view becomes empty, selected tab change, URL stays the same (expected: URL and page view changes as well)
  - choose NO
  -> selected tab updates, URL stays the same (expected: no update to the page)
